src/Reflection/InitializerExprTypeResolver.php

```diff
@@ -1726,11 +1726,17 @@ static function (Type $type, callable $traverse): Type {
 					if ($type instanceof TemplateType && !$type instanceof TypeWithClassName) {
 						return new GenericClassStringType($type);
 					} elseif ($type instanceof TypeWithClassName) {
+						$stringType = new GenericClassStringType($type);
+
 						$reflection = $type->getClassReflection();
 						if ($reflection !== null && $reflection->isFinalByKeyword()) {
-							return new ConstantStringType($reflection->getName(), true);
+							$stringType = new ConstantStringType($reflection->getName(), true);
```

Comment:
`ConstantStringType` is already considered to be a literal string, so intersecting with `AccessoryLiteralStringType` will just remove the `AccessoryLiteralStringType` again.